src/units/abilities.cpp

```diff
@@ -1667,10 +1658,29 @@ bool attack_type::special_active_impl(
 			return false;
 	}
 
+	//Add wml filter if "backstab" attribute used.
+	if (!special["backstab"].blank()) {
+		deprecated_message("backstab= in weapon specials", DEP_LEVEL::INDEFINITE, "", "Use [filter_opponent] with a formula instead; the code can be found in data/core/macros/ in the WEAPON_SPECIAL_BACKSTAB macro.");
+	}
+	config cfg = special;
+	if(special["backstab"].to_bool()){
+		config filter;
+		const std::string& backstab_formula = "	enemy_of(self, flanker) and not flanker.petrified	where	flanker = unit_at(direction_from(loc, other.facing))	";
+		filter["formula"] = backstab_formula;
+		if(!special.child("filter_opponent")){
+			cfg.add_child("filter_opponent", filter);
+		} else {
+			config filter_child = cfg.child("filter_opponent");
+			filter_child.add_child("and", filter);
+			cfg.child("filter_opponent") = filter_child;
```

Comment:
I think instead of having this line assigning the config back, you can just add an & a few lines up:

```suggestion
			config& filter_child = cfg.child("filter_opponent");
			filter_child.add_child("and", filter);
```

Reply:
Should this (both @newfrenchy83's code and the suggested replacement) use `child_or_add()` instead of `child()`?